src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -798,17 +864,243 @@ def __try_add_error(error_list, detail):
-        message = "{0} error/s reported.".format(error_count_by_operation)
-        message += " The latest {0} error/s are shared in detail. To view all errors, review this log file on the machine: {1}".format(len(errors_by_operation), self.__log_file_path) if error_count_by_operation > 0 else ""
+        success_code = Constants.PatchOperationTopLevelErrorCode.SUCCESS
+        error_code = Constants.PatchOperationTopLevelErrorCode.ERROR
+        error_count = error_count_by_operation
+        code = success_code if error_count_by_operation == 0 else error_code
+
+        # Update the errors json to include truncation detail
+        if truncated:
+            error_count += 1    # add 1 because of truncation
+            code = Constants.PatchOperationTopLevelErrorCode.WARNING if code != error_code else error_code
+
+        message = "{0} error/s reported.".format(error_count)
+        message += " The latest {0} error/s are shared in detail. To view all errors, review this log file on the machine: {1}".format(len(errors_by_operation), self.__log_file_path) if error_count > 0 else ""
         return {
-            "code": Constants.PatchOperationTopLevelErrorCode.SUCCESS if error_count_by_operation == 0 else Constants.PatchOperationTopLevelErrorCode.ERROR,
+            "code": code,
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
+        if not len(self.__assessment_packages_removed) == 0:
+            self.composite_logger.log_debug("Packages removed from assessment packages list: {0}".format(self.__assessment_packages_removed))
+        if not len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("Packages removed from installation packages list: {0}".format(self.__installation_packages_removed))
+        if len(self.__assessment_packages_removed) == 0 and len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("No packages truncated")
+
+    def __check_truncation_time_passed_by_x_sec(self, track_truncation_timestamp):
+        """ check current sys time is more than truncation time stamp by x constant seconds"""
+        if track_truncation_timestamp is None:
+            track_truncation_timestamp = datetime.datetime.now()
+
+        curr_timestamp = datetime.datetime.now()
+        return (curr_timestamp - track_truncation_timestamp).total_seconds() >= Constants.StatusTruncationConfig.SKIP_TRUNCATION_LOGIC_IN_X_SEC
```

Comment:
why is this skip truncation by secs added?

Also, the function name again does not clearly state what is happening within

Reply:
add description, done.

Reply:
Not a blocker for the first validation round: But the description, within function or in PR, does not explain "why" this skip truncation logic is added.

Also, based on my assumption, of this logic, truncation should be skipped if the remaining time (i.e. current - track_tuncation) <= Constants.StatusTruncationConfig.SKIP_TRUNCATION_LOGIC_IN_X_SEC. Is this correct?

So the last statement should be:
 return (curr_timestamp - track_truncation_timestamp).total_seconds() > Constants.StatusTruncationConfig.SKIP_TRUNCATION_LOGIC_IN_X_SEC